connector-agent-python/src/lib.rs

```diff
@@ -72,3 +76,78 @@ fn read_pg(py: Python, conn: &str, sqls: Vec<String>, schema: &str) -> PyResult<
         .collect();
     PyResult::Ok(ret.into_py_dict(py).to_object(py))
 }
+
+fn index_query(query: &str, col: &str, lower: i64, upper: i64) -> String {
+    trace!("Incoming query: {}", query);
+
+    let dialect = PostgreSqlDialect {};
+
+    let mut ast = Parser::parse_sql(&dialect, query).unwrap();
+
+    match &mut ast[0] {
+        Statement::Query(q) => match &mut q.body {
+            SetExpr::Select(select) => {
+                let cur_selection = select.selection.as_ref();
+                let mut _partition_query = format!("{} >= {} and {} < {}", col, lower, col, upper);
+                if !cur_selection.is_none() {
+                    _partition_query = format!(
+                        "{} and {} >= {} and {} < {}",
```

Comment:
Maybe we need to add bracket here. If the query is: `SELECT * FROM table WHERE a > 0 OR b < 10`, then this will make the partition query be: `SELECT * FROM table WHERE a > 0 OR b < 10 AND col >= upper AND col < lower`, which will be the same with `(a > 0) OR (b < 10 AND col >= upper AND col < lower)` since `And` has precedence over `Or` in sql. However what we actually want here is `(a > 0 OR b < 10) AND (col >= upper AND col < lower)`.